### ISSUE TITLE
SC2016 shouldn't fire on escaped backticks

### ISSUE THREAD

**Issue by dgholz** [koalaman/shellcheck#2277]
#### For bugs
- Rule Id: SC2016
- My shellcheck version (`shellcheck --version` or "online"): `0.7.2`
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2016)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash
echo hello | sed -e's/.*/`&`/'

```

#### Here's what shellcheck currently says:
`SC2016: Expressions don't expand in single quotes, use double quotes for that.`

#### Here's what I wanted or expected to see:
In this case, I see how `shellcheck` can't determine if I wanted to output literal backticks or interpolate the output of a command called `&`. So I tried explicit escaping:
```
echo hello | sed -e's/.*/\`&\`/'
```
and see the same result.

I can see how to avoid the check by replacing `` ` `` with ``'"\`"'`` but that's a bit noisy. My use case here is preparing some Markdown text and adding code formatting to some words

**Comment by koalaman** [comment]
If you are willing to add extra escaping to the backticks, then you might as well switch to double quotes throughout in this case:

    echo hello | sed -e "s/.*/\`&\`/"

However, it is indeed unfortunate more generally. GNU `sed` does ignore additional escaping in the replacement in this case, but most programs don't, and even putting the `` \` `` in the pattern would be enough to break it here so that's not a great option.

I think your original version is the best one. It would be tragic and ironic if ShellCheck made you write worse code, so please don't hesitate to [ignore](https://github.com/koalaman/shellcheck/wiki/Ignore) any issue command/file/project-wide when it doesn't add value.
